src/Compilers/Core/Portable/Compilation/NullableContextOptions.cs

```diff
@@ -0,0 +1,53 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+#nullable enable
+using System;
+
+namespace Microsoft.CodeAnalysis
+{
+    /// <summary>
+    /// Represents the default state of nullable analysis in this compilation.
+    /// </summary>
+    [Flags]
+    public enum NullableContextOptions
+    {
+        /// <summary>
+        /// The nullable analysis feature is disabled.
+        /// </summary>
+        Disable = 0,
+
+        /// <summary>
+        /// Nullable warnings are enabled and will be reported by default.
+        /// </summary>
+        Warnings = 1,
+
+        /// <summary>
+        /// Nullable annotations are enabled and will be shown when APIs
+        /// defined in this project are used in other contexts.
+        /// </summary>
+        Annotations = 1 << 1,
+
+        /// <summary>
+        /// The nullable analysis feature is fully enabled.
+        /// </summary>
+        Enable = Warnings | Annotations,
+    }
+
+    public static class NullableContextOptionsExtensions
```

Comment:
>NullableContextOptionsExtensions [](start = 24, length = 32)

not sure anyone needs the extensions on NullableContextOptions

Reply:
Could you comment on thread? For instance, does IDE need such extensions?
It feels like NullableContextOptions will be made rather than read, whereas NullableContext will be read (rather than made). That's why I'm (mildly) pushing back on these extensions.

---
In reply to: [303187463](https://github.com/dotnet/roslyn/pull/37017#discussion_r303187463) [](ancestors = 303187463)

Reply:
I'd prefer to keep the extensions for consistency between these APIs. I can see analyzers especially gating some more complicated analyses based on whether or not the compilation as a whole has nullable analysis turned on.